roles/splunk/tasks/remove_splunk.yml

```diff
@@ -11,4 +11,120 @@
     state: absent
   become: true
 
-# Todo: Remove user, group, THP/ulimit files, check for cron jobs/scripts and remove those if present
+- name: Block for non-root splunk user removal
+  block:
+    - name: Remove nix splunk user
+      ansible.builtin.user:
+        name: "{{ splunk_nix_user }}"
+        groups: "{{ splunk_nix_group }}"
+        state: absent
+        local: "{{ local_os_user }}"
+      become: true
+
+    - name: Remove nix splunk group
+      ansible.builtin.group:
+        name: "{{ splunk_nix_group }}"
+        state: absent
+        local: "{{ local_os_group }}"
+      become: true
+  when: splunk_nix_user != 'root'
+
+- name: Remove Disable-THP/Ulimit files
+  block:
+    - name: Stop Disable-THP service
```

Comment:
Sorry for the nit-pick, but this service does not need to be stopped, because the exec exits right after it runs, so removing the unit file and reloading systemd should be enough.
Also, the conditional for removing the unit file should be:
```
      when:
        - not use_tuned_thp
        - ansible_connection!="docker"
        - "'full' in group_names"
```

Reply:
Done - fixed in [d31f002](https://github.com/splunk/ansible-role-for-splunk/pull/161/commits/d31f00288c5dbbb974b508df1cb30175f8c5443d)